src/main/scala/workflow/Estimator.scala

```diff
@@ -1,21 +1,34 @@
-package pipelines
+package workflow
 
 import java.io.Serializable
 
 import org.apache.spark.rdd.RDD
 
+import scala.reflect.ClassTag
+
 /**
  * An estimator has a `fit` method which takes an input and emits a [[Transformer]]
  * @tparam A The type of input this estimator (and the resulting Transformer) takes
  * @tparam B The output type of the node this estimator produces when being fit
  */
-abstract class Estimator[A, B] extends Serializable {
+abstract class Estimator[A, B] extends EstimatorNode  {
   /**
-   * An estimator has a `fit` method which emits a pipeline node.
+   * An estimator has a `fit` method which emits a [[Transformer]].
    * @param data Input data.
-   * @return A PipelineNode (Transformer) which can be called on new data.
+   * @return A [[Transformer]] which can be called on new data.
    */
-  def fit(data: RDD[A]): Transformer[A, B]
+  protected def fit(data: RDD[A]): Transformer[A, B]
+
+  private[workflow] final def fit(dependencies: Seq[RDD[_]]): TransformerNode[_] = fit(dependencies.head.asInstanceOf[RDD[A]])
+
+  def withData(data: RDD[A]): Pipeline[A, B] = {
```

Comment:
So this is the thing that constructs a pipeline from nothing in case you ant just an estimator pipeline? Can we get a comment to that effect?